source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -592,8 +491,20 @@ namespace aspect
                   corners[1][0] = corners[2][0];
                   corners[3][0] = corners[0][0];
                 }
-              corners_rectangle = corners;
             }
+          // Check that the calculated corners also obey the rules for the location of the coners.
+          AssertThrow (corners[0][0] >= corners[1][0],
+                       ExcMessage ("The longitude of the NE corner (" + boost::lexical_cast<std::string>(corners[0][0]) + ") cannot be smaller then the longitude of the NW corner (" + boost::lexical_cast<std::string>(corners[1][0]) + "). This is an internal check, if you see this please contact the developer."));
+
+          AssertThrow (corners[0][1] >= corners[3][1],
+                       ExcMessage ("The latitude of the NE (" + boost::lexical_cast<std::string>(corners[0][1]) + ") corner cannot be larger then the longitude of the SE corner (" + boost::lexical_cast<std::string>(corners[3][1]) + "). This is an internal check, if you see this please contact the developer."));
+
+          AssertThrow (corners[2][0] <= corners[3][0],
+                       ExcMessage ("The longitude of the SW (" + boost::lexical_cast<std::string>(corners[2][0]) + ") corner cannot be larger then the longitude of the SE corner (" + boost::lexical_cast<std::string>(corners[3][0]) + "). This is an internal check, if you see this please contact the developer."));
+
+          AssertThrow (corners[2][1] <= corners[1][1],
+                       ExcMessage ("The latitude of the SW corner (" + boost::lexical_cast<std::string>(corners[2][1]) + ") cannot be smaller then the longitude of the NW corner (" + boost::lexical_cast<std::string>(corners[1][1]) + "). This is an internal check, if you see this please contact the developer."));

```

Comment:
If these are internal checks that should only fail if the above formula is wrong or somebody fiddles with the code, then they should be Assert instead of AssertThrow. Not very important in this case though.